No Walls, Only Mountain - Bastion Gate Remake + Town Fixes

ophaq opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

## About The Pull Request

Adds the following:
- A new towner house near the well behind smithing guild with a scenic view and inside the mountain.
- The towner house near the church that looked shitty near the stairs? Has a little cave entrance now. It leads nowhere but it is neat. Feels more like a slummy towner house.
- Removed scholar's closet starting essences and drugs in trade for a westleach and swampleaf so they can consider gardening or giving it to soilsons. The court physician gets these and it gives them something to on the side. The fish and meat essences stayed though.
- Numerous small fixes and changes, such as merchant window fixes and removing an extra vomitorium in scholar lab.
- Many area reconversions to mountain area as it should be done with this new update adding more mountain. Some were wrong too so I took the liberty to fix those up. Only thing this affects is sound effects and game recognizing it is a 'mountain' zone for those curious.
- Some fixes in the druid grove, aesthetics like will-o-wisp mote lights/waterfall changes.

Biggest Change:
- Bastion walls are now mountain. The surrounding mountains now 'hug' the bastion gate towers as if the town was built into a large alcove within a mountain. Although we still have the ocean nearby to the north so not quite like that but you get the picture.
- No more pit, and a single gate at the front that is PREOPENED round start. Towners rejoice, no more waiting for people to roll it up on lowpop or because they're lazy.

## Why It's Good For The Game
I've heard of people voting or suggesting this. There was a PR the other day that did this poorly so I took some time out of my day to do a proper looking version. You can dig around town now if you want to slip by sneakily. Gate sucked anyway and a lot of people hated it, but at least it's only one now and no more autism pit that didn't get used. You can properly siege the town now without falling in said pit.

## Proof of Testing (Required)
Main changes, should capture a lot of what I did:
![bastion1](https://github.com/user-attachments/assets/18c2a527-93ca-4763-bc73-81a0954b5a31)
![bastion2](https://github.com/user-attachments/assets/1f9c9431-024b-46c8-9f44-2f4b85aa1ab1)
![bastion3](https://github.com/user-attachments/assets/03896577-0f15-4f0b-980b-abda48050362)

notafurryy commented:
What was bad about having those walls open though?

ophaq commented:
> What was bad about having those walls open though? On the top level atleast

The roof? Didn't make sense to have a roof on top of the mountain to me.

notafurryy commented:
> The roof? Didn't make sense to have a roof on top of the mountain to me.

It could just be w carved peak?

ophaq commented:
> > The roof? Didn't make sense to have a roof on top of the mountain to me.
> 
> It could just be w carved peak?

I guess but is it really going to matter having a roof that isn't even that wide and there is just plain thick mountain below? Might as well block it off. The gate area still has a roof and it isn't like people are going to be climbing the sides where the mountains are. If anything they'll be digging into town to get in past the gate. That is, if the gate isn't just sitting open.

notafurryy commented:
> I guess but is it really going to matter having a roof that isn't even that wide and there is just plain thick mountain below? Might as well block it off. The gate area still has a roof and it isn't like people are going to be climbing the sides where the mountains are. If anything they'll be digging into town to get in past the gate. That is, if the gate isn't just sitting open.

The town currently has sets of buildings on each end of the gate that let you scale it and get on top

ophaq commented:
> > I guess but is it really going to matter having a roof that isn't even that wide and there is just plain thick mountain below? Might as well block it off. The gate area still has a roof and it isn't like people are going to be climbing the sides where the mountains are. If anything they'll be digging into town to get in past the gate. That is, if the gate isn't just sitting open.
> 
> The town currently has sets of buildings on each end of the gate that let you scale it and get on top of it.

![1](https://github.com/user-attachments/assets/c4e8b709-fe64-4db7-8fc7-d3cf26a6c890)
True. There is a regular way you can still get on the roof by running and jumping on the left of the image but I added a cavern for an alternative route.

NikolayByondPlayer commented:
Noice

Kregle commented:
Put the gate on the inside so the Bastion still has the option of tolls

ophaq commented:
> Put the gate on the inside so the Bastion still has the option of tolls

Good idea.

Also this PR is going to delay the merging of this and I will likely have to copy/paste the new changes onto my current changes so now we have to wait for it to be full merged...

https://github.com/Rotwood-Vale/Ratwood-Keep/pull/2213

ophaq commented:
Added some smoke effects at the waterfall and fixed some of the flow, added something like a zen sitting spot near the waterfall. Smoke effect was increased to like 14000 or something, which is like four hours. If that somehow affects game performance server side I'll remove them but I don't think they will. I experienced no lag sitting at them for a while in testing. Also added some mote light will o wisps around the grove.

![2](https://github.com/user-attachments/assets/ed5297d8-1d1b-452b-aa7c-312df2a1dcff)

hugmeorelse commented:
You're a beast.

ophaq commented:
- Added back barkeep/cook rooms but swankier.
- Bigger whitevein stair entrance. Hopefully more noticeable.
- 30% less town lamps, town will be darker in spots except main road leading to keep.
![white](https://github.com/user-attachments/assets/3d849578-664c-4eb5-b14a-f1dd08c7a6f5)
![tav](https://github.com/user-attachments/assets/5a030f4b-0da2-48e9-afe0-f690ca460e3a)

ophaq commented:
Currently trying to figure out how to get the new mage stuff in to update my map so it has the tower. Strong DMM won't recognize the new tiles and items despite syncing my fork but I'll post when this is ready with the new mage stuff.

ophaq commented:
OK mage tower stuff is implemented in to the new changes and I fixed up some stuff with it. Should be good to go.

ophaq commented:
Adds in this:
https://github.com/Rotwood-Vale/Ratwood-Keep/pull/2438

Also made whitevein entrance less like a sewer and more like closer to the sewer. Fresh water flows under the keep's moat now but there is still some outflow sewage pipes.

![bog](https://github.com/user-attachments/assets/27630711-0d01-4f87-a5fb-239762d07637)

ophaq commented:
Added:
- Handgonne back in bastion tower's warden office.
- 3 arquebus in keep armory.

ophaq commented:
Church has a small vault now behind the room where you get the amulets (This was a request due to church constantly being poor when in fact it should have some starting money). It has some starting money in it but not that much. Also added a scom to the garden so people there can listen to what is happening, and also fixed up some windows in town to be opened from within.

notafurryy commented:
Could you cut down the ammount of kneestingers in the sewers? There's way too many of them and the beggars/spiders keep killing themselves on them

ophaq commented:
> Could you cut down the ammount of kneestingers in the sewers? There's way too many of them and the beggars/spiders keep killing themselves on them

What we need to do is make it so that simple mobs have stinger immunity trait or the AI to path around it.

Alanii commented:
Another request
- Swap the farm and carriage locations (Place carriage north of farm)
- Remove the farm fortress elements.
- Gut the farm animal barn and just replace it with a chicken coop
- Remove some of the excess mountain WEST of low-town 
- Move the farm animals to a sort of "open range" field seperated as needed to prevent the Saiga, Goats and Cows from killing each other.

That is really the only thing I would like changed, outside of moving the vanguard out of the "town gate" and into a proper outpost structure within low-town.

ophaq commented:
> Another request
> 
>     * Swap the farm and carriage locations (Place carriage north of farm)
> 
>     * Remove the farm fortress elements.
> 
>     * Gut the farm animal barn and just replace it with a chicken coop
> 
>     * Remove some of the excess mountain WEST of low-town
> 
>     * Move the farm animals to a sort of "open range" field seperated as needed to prevent the Saiga, Goats and Cows from killing each other.
> 
> 
> That is really the only thing I would like changed, outside of moving the vanguard out of the "town gate" and into a proper outpost structure within low-town.

This is gonna be a different PR.

ophaq commented:
Relocated keep dungeon to below the throne room and removed the trap door and volf pit. As the volf pit was constantly abused by dukes and made for poor RP choices, this was requested so I relocated the dungeon below instead.

Apothecary has a second floor with some poison berry/westleach and a hoe (no other tools) and scholar has a hoe now too (no other tools since they have the option to garden wherever they find a spot). The apothecary also has a shared quarters, supporting up to two doctors in the house. Also fixed up a few areas that I missed before and a stray mossy wall in the mine upper level I found while mining once.

Pictures (slightly outdated but w/e):
![apoth1](https://github.com/user-attachments/assets/b89f2396-709d-47b0-b5df-15ee2ff088e8)
![apoth2](https://github.com/user-attachments/assets/1300e4e3-3ef6-4f53-85d0-87f2859e931f)
![keepdungeon](https://github.com/user-attachments/assets/1829d413-6732-40c5-95b6-74bf5f4fdc18)

notafurryy commented:
> What we need to do is make it so that simple mobs have stinger immunity trait or the AI to path around it

![image](https://github.com/user-attachments/assets/235c58eb-0dcf-4477-b592-7ce38aabd9cb)
the ammount of stingers is nuts though, as if the leeches enemies and mud wasn't enough

ophaq commented:
Closing this as the other PR is the same and will be full merged.
